proxy_map.go

```diff
@@ -38,43 +38,36 @@ Map is a distributed map.
 Hazelcast Go client enables you to perform operations like reading and writing from/to a Hazelcast Map with methods like Get and Put.
 For details, see https://docs.hazelcast.com/imdg/latest/data-structures/map.html
 
-Listening for Map Entry Events
+Listening for Map Events
 
-The first step of listening to entry-based events is  creating an instance of MapEntryListenerConfig.
-MapEntryListenerConfig contains options to filter the events by key and/or predicate and has an option to include the value of the entry, not just the key.
-You should also choose which type of events you want to receive.
-In the example below, a listener configuration for added and updated entries is created.
-Entries only with key "somekey" and matching to predicate year > 2000 are considered:
+To listen events of a map, you can use the AddListener, AddListenerWithKey, AddListenerWithPredicate and AddListenerWithPredicateAndKey methods.
+The first method adds a listener to the map's all events. The others filter the events depending on a key and/or a predicate.
+In all methods you specify whether you want to include value in the event or not.
+
+You can pass a MapListener struct to these methods to add handlers to different event types. You can add different handlers to different event types
+with a single MapListener struct. If you don't specify a handler in for an event type in MapListener struct, there will be no handler for that event.
+In the example below, a listener for added and updated entry events is created. Entries only with key "somekey" and matching to predicate year > 2000
+are considered:
 
 	entryListenerConfig := hazelcast.MapEntryListenerConfig{
 		Key: "somekey",
 		Predicate: predicate.Greater("year", 2000),
 		IncludeValue: true,
 	}
-	entryListenerConfig.NotifyEntryAdded(true)
-	entryListenerConfig.NotifyEntryUpdated(true)
-	m, err := client.GetMap(ctx, "somemap")
-
-After creating the configuration, the second step is adding an event listener and a handler to act on received events:
 
-	subscriptionID, err := m.AddEntryListener(ctx, entryListenerConfig, func(event *hazelcast.EntryNotified) {
-		switch event.EventType {
-		case hazelcast.EntryAdded:
+	m, err := client.GetMap(ctx, "somemap")
```

Comment:
Could you add a line to indicate error checking is missing, maybe something like:
```
// error checking is omitted.
```
Probably after also the `subscriptionID, err ...` line below.